dockstore-webservice/src/main/java/io/dockstore/webservice/resources/UserResource.java

```diff
@@ -443,25 +440,29 @@ public User removeUserFromGroup(@ApiParam(hidden = true) @Auth User authUser,
     @UnitOfWork
     @Path("/starredTools")
     @ApiOperation(value = "Get the logged-in user's starred tools", response = Entry.class, responseContainer = "List")
-    public Set<Entry> getStarredTools(@ApiParam(hidden = true) @Auth User user) {
+    public SortedSet<Entry> getStarredTools(@ApiParam(hidden = true) @Auth User user) {
         User u = userDAO.findById(user.getId());
-        Comparator<Entry> byEntryId = Comparator.comparingLong(Entry::getId);
-        Supplier<TreeSet<Entry>> supplier = () -> new TreeSet<>(byEntryId);
-        return u.getStarredEntries().stream().filter(element -> element instanceof Tool).sorted(byEntryId)
-                .collect(Collectors.toCollection(supplier));
+        return filter(u.getStarredEntries(), Tool.class);
     }
 
     @GET
     @Timed
     @UnitOfWork
     @Path("/starredWorkflows")
     @ApiOperation(value = "Get the logged-in user's starred workflows", response = Entry.class, responseContainer = "List")
-    public Set<Entry> getStarredWorkflows(@ApiParam(hidden = true) @Auth User user) {
+    public SortedSet<Entry> getStarredWorkflows(@ApiParam(hidden = true) @Auth User user) {
         User u = userDAO.findById(user.getId());
-        Comparator<Entry> byEntryId = Comparator.comparingLong(Entry::getId);
-        Supplier<TreeSet<Entry>> supplier = () -> new TreeSet<>(byEntryId);
-        return u.getStarredEntries().stream().filter(element -> element instanceof Workflow).sorted(byEntryId)
-                .collect(Collectors.toCollection(supplier));
+        return filter(u.getStarredEntries(), Workflow.class);
+    }
+
+    private SortedSet<Entry> filter(SortedSet<Entry> set, Class someClass) {
+        SortedSet filteredSet = new TreeSet();
+        for (Entry entry : set) {
+            if (someClass.isInstance(entry)) {
+                filteredSet.add(entry);
+            }
+        }
+        return filteredSet;
```

Comment:
Originally, I was using lambda with collect and it didn't retain the order.  I suppose I can change it back to lambda.